frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/desktop_toolbar/link/link_extension.dart

```diff
@@ -8,32 +8,65 @@ extension LinkExtension on EditorState {
     if (node == null) {
       return;
     }
+    final attributes = _getAttribute(node, selection);
```

Comment:
I think we're not getting all the attributes from the selection. For example:

Before:

_http_://**appflowy**.io <- (add one more o here)

After:

_http_://**appflowy**.ioo The result should be ioo without extra formatting.

We have a slice attributes function that gets the attributes based on the index. `appflowyEditorSliceAttributes`

Reply:
yes, the logic about getting Attributes is not correct, I need to adjust it